features/netsocket/UDPSocket.h

```diff
@@ -24,11 +24,12 @@
 #include "netsocket/NetworkStack.h"
 #include "netsocket/NetworkInterface.h"
 #include "rtos/EventFlags.h"
+#include "RAWIPSocket.h"
 
 
 /** UDP socket implementation.
  */
-class UDPSocket : public InternetSocket {
+class UDPSocket : public RAWIPSocket {
```

Comment:
The code sharing is good, but the class heirarchy is off. This lets you pass a `UDPSocket *` to someone expecting a `RAWIPSocket *`. Previous discussion suggested inserting an underlying datagram socket base class.